apisix/control/v1.lua

```diff
@@ -119,11 +201,19 @@ local function iter_and_find_healthcheck_info(values, src_type, src_id)
 
     for _, value in core.config_util.iterate_values(values) do
         if value.value.id == src_id then
-            if not value.checker then
+            local checks = value.value.checks or
+                (value.value.upstream and value.value.upstream.checks)
+            if not checks then
                 return nil, str_format("no checker for %s[%s]", src_type, src_id)
             end
 
-            return extra_checker_info(value, src_type)
+            local info = extra_checker_info(value)
+            if checks.active and checks.active.type then
+                info.type = checks.active.type
+            elseif checks.passive and checks.passive.type then
+                info.type = checks.passive.type
+            end
```

Comment:
Can this duplicate code be extracted into a function?

Reply:
Done.